pkg/metrics/metrics.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	kubefedClustersNotReadyCount = prometheus.NewCounter(
```

Comment:
These are counters so only monotonically increasing - don't we want to use a gauge for these metrics?

Reply:
Yeah, that sounds like a better approach. I will change it